### PR TITLE
src: emmc: Add configurable size alignment for expanding partitions

### PR DESCRIPTION
**Pull request by landerweit-phytec**
Add size alignment for expanding partitions. The alignment can be
configured with for each partition individually with 'align'. The
default alignment is 1KiB.

### CODE REVIEW

File: doc/layout-config.rst [hunk 1]
```diff
@@ -59,6 +59,9 @@ options:
 ``offset`` (integer/string)
    The offset of a partition.
 
+``align`` (integer/string)
+   Aligns the size of the partition to the specified value. The default is 1KiB.
```
Comment (mschwan-phytec) [inline doc/layout-config.rst:63]: I think we should be more precise regarding the default. It is actually two sectors, which in almost all cases is 1KiB (2 × 512B), but I still think we should mention it that way.

File: src/emmc.c [hunk 2]
```diff
@@ -186,15 +187,19 @@ pu_emmc_setup_layout(PuFlash *flash,
...
+        }
+
         /* Increase size of logical partitions by 2 sectors for EBR.
          * The 2 sectors are removed during partition creation. */
         if (part->type == PED_PARTITION_LOGICAL) {
             part->size += 2;
         }
 
-        g_debug("%s: type=%d filesystem=%s start=%lld size=%lld offset=%lld expand=%s",
+        g_debug("%s: type=%d filesystem=%s start=%lld size=%lld offset=%lld align=%lld expand=%s",
```
Comment (mschwan-phytec) [inline src/emmc.c:200]: I think it is time to break this string into two lines. The soft limit of 80 characters has been stretched enough. :)

File: src/emmc.c [hunk 3]
```diff
@@ -615,9 +621,9 @@ pu_emmc_parse_partitions(PuEmmc *emmc,
             return FALSE;
         }
 
-        g_debug("Parsed partition: label=%s filesystem=%s type=%s size=%lld offset=%lld expand=%s",
+        g_debug("Parsed partition: label=%s filesystem=%s type=%s size=%lld offset=%lld align=%lld expand=%s",
```
Comment (mschwan-phytec) [inline src/emmc.c:624]: Same for this one. Breaking this string up into two lines would most certainly benefit its readability.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
I am unsure if we should rather rename the option to "block-size" instead of "align". Looking at the man pages of e.g. mkfs.ext4 shows that *alignment* means the positioning of a partition to certain blocks of sectors/bytes. In contrast, *block size* means a multiplication factor the partition size has to fit in, but with no limits to the start and end alignment.

**Comment by landerweit-phytec** [comment]
Renamed the configuration option from 'align' to 'block-size'.

File: src/emmc.c [hunk 4]
```diff
@@ -186,15 +187,20 @@ pu_emmc_setup_layout(PuFlash *flash,
...
+
         /* Increase size of logical partitions by 2 sectors for EBR.
          * The 2 sectors are removed during partition creation. */
         if (part->type == PED_PARTITION_LOGICAL) {
             part->size += 2;
         }
 
-        g_debug("%s: type=%d filesystem=%s start=%lld size=%lld offset=%lld expand=%s",
+        g_debug("%s: type=%d filesystem=%s start=%lld size=%lld offset=%lld \
+                block-size=%lld expand=%s",
```
Comment (mschwan-phytec) [inline src/emmc.c:201]: Multiline strings should be split like the following:
```
"one "
"two"
```
Using `\` will carry the indentation of the next line into the string, which we do not want.

File: doc/layout-config.rst [hunk 5]
```diff
@@ -59,6 +59,10 @@ options:
 ``offset`` (integer/string)
    The offset of a partition.
 
+``block-size`` (integer/string)
+   Set the partition size to a multiple of the specified value. The
+   default is 2 sectors, which is equal to 1KiB.
```
Comment (mschwan-phytec) [inline doc/layout-config.rst:64]: Please use the following term here:
```
[...], which is almost always equal to 1KiB.
```

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**